lib/encoder.py

```diff
@@ -813,9 +815,15 @@ def ReadEncodingResult(self, encoding, scoredir=None):
       with open(filename, 'r') as resultfile:
```

Comment:
I suggest moving this into a helper method and give it a good name (or a comment) to explain what's going on, like _ReadAsJsonOrPython.